src/storage-operators/src/s3_oneshot_sink.rs

```diff
@@ -45,21 +47,33 @@ pub fn copy_to<G, F>(
     connection_details: S3UploadInfo,
     connection_context: ConnectionContext,
     aws_connection: AwsConnection,
+    sink_id: GlobalId,
     connection_id: GlobalId,
     onetime_callback: F,
 ) where
     G: Scope<Timestamp = Timestamp>,
     F: FnOnce(Result<u64, String>) -> () + 'static,
 {
-    let scope = input_collection.scope();
+    let mut scope = input_collection.scope();
     let worker_id = scope.index();
+    let is_leader = usize::cast_from((sink_id, "leader").hashed()) % scope.peers() == worker_id;
```

Comment:
nit (you can ignore): factor the `RawSourceCreationConfig::responsible_for` method out as a function to be used here